src/vcpkg/base/downloads.cpp

```diff
@@ -447,31 +450,58 @@ namespace vcpkg::Downloads
             const auto dir = download_path_part_path.parent_path();
             fs.create_directories(dir, VCPKG_LINE_INFO);
 
-            auto f = fs.open_for_write(download_path_part_path, VCPKG_LINE_INFO);
-
             const auto sanitized_url = replace_secrets(url, secrets);
             Debug::print("Downloading ", sanitized_url, "\n");
             static auto s = WinHttpSession::make().value_or_exit(VCPKG_LINE_INFO);
-            auto conn = WinHttpConnection::make(s.m_hSession.get(), hostname, port);
-            if (!conn)
+            for (size_t trials = 0; trials < 4; ++trials)
             {
-                Strings::append(errors, sanitized_url, ": ", conn.error(), '\n');
-                return false;
-            }
-            auto req = WinHttpRequest::make(
-                conn.get()->m_hConnect.get(), split_uri.path_query_fragment, split_uri.scheme == "https");
-            if (!req)
-            {
-                Strings::append(errors, sanitized_url, ": ", req.error(), '\n');
-                return false;
-            }
-            auto forall_data = req.get()->forall_data([&f](Span<char> span) { f.write(span.data(), 1, span.size()); });
-            if (!forall_data)
-            {
-                Strings::append(errors, sanitized_url, ": ", forall_data.error(), '\n');
-                return false;
+                if (trials > 0)
+                {
+                    // 0.5s, 1s, 2s
+                    Debug::print("Download failed -- retrying after ", 250 << trials, " ms.\n");
```

Comment:
It looks to me like you're retrying after 0.25s, 0.5s, 1s here, not 0.5s, 1s, 2s as intended.

Reply:
I wonder if even your intended intervals are too aggressive.  Something more like 2s, 4s, 8s sounds more reasonable to me if what we're seeing is momentary high network load.

Reply:
Oops, I was wrong about your intervals not being what you intended, but the other point remains.